_posts/2019-03-14-2019-kafka-connector.md

```diff
@@ -0,0 +1,194 @@
+    handler: ./kafka-message
+    image: <your_dockerhub_username>/kafka-message:latest
+    annotations:
+      topic: faas-request
+```
+
+Build, Push and Deploy the function with single command:
+
+```bash
+$ faas-cli up
+```
+
+Voila, your function is now subscribed to the topic and it can be invoked by producing message on that topic.
+
+Now you might ask:
+
+"Great, but how am I supposed see the output of my function?"
+
+Remember in the configuration we set `print_response_body` to `true` so we can see the response body in the logs of the Kafka Connector.
+
+## Generate load on a topic
+
+Lets proceed by opening two terminals, one will be used to check the output of the function and the other will produce load on the topic.
+
+In the first terminal open shell session inside broker pod to connect with the producer:
+
+```bash
+$ kubectl exec -n openfaas -t -i $(kubectl get pods -n openfaas -l component=kafka-broker -o name|cut -d'/' -f2) -- /opt/kafka_2.12-0.11.0.1/bin/kafka-console-producer.sh --broker-list kafka:9092 --topic faas-request
+```
+
+In the second terminal follow the Kafka Connector logs with this command:
+
+```bash
+$ kubectl logs $(kubectl get pods -o name -n openfaas | grep -m1 kafka-connector | cut -d'/' -f 2) -n openfaas --tail 100 -f
+```
+
+In the first terminal write the message you would like your function to receive and check the logs output.
+
+```
+...
+2019/03/16 10:08:26 Syncing topic map
+[#1] Received on [faas-request,0]: 'Test the function.'
+2019/03/16 10:08:26 Invoke function: figlet
+2019/03/16 10:08:27 connector-sdk got result: [200] faas-request => figlet (480) bytes
+[200] faas-request => figlet
+ _____         _     _   _             __                  _   _               
+|_   _|__  ___| |_  | |_| |__   ___   / _|_   _ _ __   ___| |_(_) ___  _ __    
+  | |/ _ \/ __| __| | __| '_ \ / _ \ | |_| | | | '_ \ / __| __| |/ _ \| '_ \   
+  | |  __/\__ \ |_  | |_| | | |  __/ |  _| |_| | | | | (__| |_| | (_) | | | |_ 
+  |_|\___||___/\__|  \__|_| |_|\___| |_|  \__,_|_| |_|\___|\__|_|\___/|_| |_(_)
+                                                                               
+
+2019/03/16 10:08:29 Syncing topic map
+2019/03/16 10:08:32 Syncing topic map
+2019/03/16 10:08:35 Syncing topic map
+...
+```
+
+## But why do I need this combination?
+
+There are multiple real world examples where Kafka in combination with the Serverless approach provided by OpenFaaS becomes really powerful tool as Kafka can make even the smallest of systems extensible and OpenFaaS provides us with the quick way to add those extensions.
+
+Lets say you need to process customer information. When customer applies for you application you need to process it and store the information in a database and there are tons of customers. Just create topic for example `user-creation` which will be populated with the information and then create OpenFaaS function which handles the processing and updating of that information to the database.
```

Comment:
Not sure about the example, I've altered a little, but I'm still not 100% with it:

When a customer creates an account you need to process their information and store it in a database.  However, there are a number of other departments also interested in new customers - theres marketing who want to send a welcome pack, there's payments who want to check their credit score. Just create topic, for example `user-creation`, which will be populated with the information and then create OpenFaaS function which handles the processing and updating of that information to the database.  Similarly marketing and payments can both implement functions to process the data in a way that makes sense for them.

Reply:
Thanks for this Richard I had problems coming up with good example

Reply:
Changed the example with yours 💯

Reply:
It is better